internal/pkg/describe/service.go

```diff
@@ -294,17 +316,39 @@ func (d *appRunnerServiceDescriber) Service() (*apprunner.Service, error) {
 	return service, nil
 }
 
+// IsPrivate returns true if the service is configured as non-public.
+func (d *appRunnerServiceDescriber) IsPrivate() (bool, error) {
+	vicArn, err := d.vpcIngressConnectionARN()
+	if err != nil {
+		return false, err
+	}
+
+	return vicArn != nil, nil
+}
+
 // ServiceURL retrieves the app runner service URL.
 func (d *appRunnerServiceDescriber) ServiceURL() (string, error) {
-	service, err := d.Service()
+	vicArn, err := d.vpcIngressConnectionARN()
 	if err != nil {
 		return "", err
 	}
 
-	return formatAppRunnerUrl(service.ServiceURL), nil
+	if vicArn != nil {
+		url, err := d.apprunnerClient.PrivateURL(context.Background(), *vicArn)
```

Comment:
This doesn't look to be the right place for creating a context. My suggestion is still we use ctx when it is *required* it'll be less painful when we switch to sdk v2.

Reply:
discussed offline, i'll just use the non-context func for now